message_ix/util/__init__.py

```diff
@@ -232,3 +233,23 @@ def expand_dims(scenario: Scenario, name, **data):
 
         # Add the expanded data
         scenario.add_par(name, new_data)
+
+
+def sankey_mapper(df, year, region, flows_not_needed=[], variables_not_needed=[]):
+    df_filtered = df.filter(region=region + "*", year=year)
+
+    mapping = {}
+    for var in df_filtered["variable"]:
```

Comment:
this block could be cleaned up as follows:

```
is_input = gvc(var, 0) == "in"
start_idx, end_idx = [1, 2], [3, 4] if is_input else [3, 4], [1, 2]
mapping[var] = (gvc(var, start_idx, join=True), gvc(var, end_idx, join=True))
```

Reply:
Thanks for this suggestion, it almost worked out of the box :)
Maybe most interesting to you: you need
```python
(start_idx, end_idx) = ([1, 2], [3, 4]) if is_input else ([3, 4], [1, 2])
```
to do the combined check you had in mind.